### PR TITLE
Make addmod and mulmod revert if the last argument is zero.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/2694

### CODE REVIEW

File: libsolidity/codegen/ExpressionCompiler.cpp [hunk 1]
```diff
@@ -765,7 +765,11 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 		case FunctionType::Kind::AddMod:
 		case FunctionType::Kind::MulMod:
 		{
-			for (unsigned i = 0; i < 3; i ++)
+			arguments[2]->accept(*this);
+			utils().convertType(*arguments[2]->annotation().type, IntegerType(256));
+			m_context << Instruction::DUP1 << Instruction::ISZERO;
+			m_context.appendConditionalRevert();
```
Comment (axic) [inline libsolidity/codegen/ExpressionCompiler.cpp]: Should this be revert or assert?

Comment (axic) [reply libsolidity/codegen/ExpressionCompiler.cpp]: div/mod uses assert: https://github.com/ethereum/solidity/blob/develop/libsolidity/codegen/ExpressionCompiler.cpp#L1494-L1496

Comment (chriseth) [reply libsolidity/codegen/ExpressionCompiler.cpp]: Ah you are right!

Comment (chriseth) [reply libsolidity/codegen/ExpressionCompiler.cpp]: Fixed.

File: Changelog.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ### 0.4.21 (unreleased)
 
 Features:
+ * Code Generator: Make ``molmod(a, b, 0)`` and ``addmod(a, b, 0)`` revert as experimental 0.5.0 feature.
```
Comment (axic) [inline Changelog.md]: Need to update all the documentation changes from `revert` to throw (or whatever wording we're using).

**Review by axic — APPROVED** [review]

**Pull request merged**